balancer/weightedroundrobin/weightedroundrobin.go

```diff
@@ -19,11 +19,40 @@
 // Package weightedroundrobin defines a weighted roundrobin balancer.
 package weightedroundrobin
 
+import (
+	"google.golang.org/grpc/resolver"
+)
+
 // Name is the name of weighted_round_robin balancer.
 const Name = "weighted_round_robin"
 
-// AddrInfo will be stored inside Address metadata in order to use weighted roundrobin
-// balancer.
+// attributeKey is the type used as the key to store AddrInfo in the Attributes
+// field of resolver.Address.
+type attributeKey struct{}
+
+// AddrInfo will be stored inside Address metadata in order to use weighted
+// roundrobin balancer.
 type AddrInfo struct {
 	Weight uint32
 }
+
+// SetAddrInfo sets addInfo in the Attributes field of addr.
+func SetAddrInfo(addrInfo *AddrInfo, addr *resolver.Address) {
```

Comment:
There's a discussion on whether the function should take a pointer and modify, or make a copy: https://github.com/grpc/grpc-go/pull/3494#discussion_r409678912. Let's wait for that.

Reply:
And make the function experimental.

Reply:
Please change these to not be pointers - do copies instead.  Since these are not performance sensitive, we will optimize for code clarity and proper usage, which copying would better provide, I believe.  E.g. using a pointer means modifications to the `AddrInfo` passed in will affect addresses which received a pointer to it in their attributes.  This could lead to bugs.

Reply:
Done.